docs/contributor/compiler_walkthrough.md

```diff
@@ -4,48 +4,56 @@ This guide will take you through all of the phases of the compiler to give you a
 
 We'll largely be following the `next_state` function in [compile.re](https://github.com/grain-lang/grain/blob/main/compiler/src/compile.re).
 
+## An overview of the compiler
+
+The grain compiler is a [multi-stage](https://en.wikipedia.org/wiki/Multi-pass_compiler) compiler, which means instead of converting directly from `grain syntax` into `wasm` we take multiple passes over the input program performing smaller transformations until we get to the final output. This approach allows us to have a more maintainable compiler and perform deeper analysis on your code which allows us to provide better errors and more abstraction.
```

Comment:
```suggestion
The Grain compiler is a [multi-stage](https://en.wikipedia.org/wiki/Multi-pass_compiler) compiler, which means instead of converting directly from Grain syntax into `wasm` code, we send the input program through multiple phases, transforming from one intermediate representation to the next until we get to the final output. This approach allows us to have a more maintainable compiler and perform deeper analysis of the source code, which lets us provide better errors and better code output.
```